### PR TITLE
Go lint fixes, go 1.18, go mod vendor, updated GH Actions

### PR DESCRIPTION
**Pull request by JanneKiiskila**
Fix `golint` findings.
Fix pysh findings on `make.sh`.
Update go version to 1.18 to align with others (and `go mod tidy` to update `go.sum`).
Bring back vendor-folder (for Yocto builds), i.e. `go mod vendor`.
Update GitHub actions to run `golint` and `pysh-check`.

### CODE REVIEW

File: make.sh [hunk 1]
```diff
@@ -26,8 +26,8 @@ run() {
 test() {
     go vet
     if [[ -n "$1" ]]; then
-        go test $1 -timeout 15s github.com/PelionIoT/pe-terminal/components
-    else
+        go test "$1" -timeout 15s github.com/PelionIoT/pe-terminal/components
+    elses
```
Comment (petedyerarm) [inline make.sh:30]:
```suggestion
    else
```

**Review by petedyerarm — CHANGES_REQUESTED** [review]

File: make.sh (hunk 1 above)
Comment (JanneKiiskila) [reply make.sh:30]: Wonder why pysh does not spot that?

Comment (JanneKiiskila) [reply make.sh]: Fixed.

**Comment by JanneKiiskila** [comment]
Builds w LmP w/o the loading exception.

**Comment by JanneKiiskila** [comment]
e2e tests also pass.

**Pull request merged**